agent/engine/docker_task_engine.go

```diff
@@ -45,6 +45,7 @@ const (
 	capabilityPrefix             = "com.amazonaws.ecs.capability."
 	capabilityTaskIAMRole        = "task-iam-role"
 	capabilityTaskIAMRoleNetHost = "task-iam-role-network-host"
+	capabilityTaskResourceLimit  = "task-resource-limit"
```

Comment:
Can you modify this to be `task-cpu-mem-limit`? `resource` has a pretty broad definition and seems like we're overloading it here. This applies to other names as well, let's make it very explicit by modifying `TaskResourceLimit` to be `TaskCPUMemLimit` everywhere.

Reply:
Given that memory is optional, should we split this up into two distinct capabilities ?

Reply:
@vsiddharth the capabilities do not reflect fields that are set in tasks. They are a way of letting agent register if it's capable of handling "certain things". In this particular scenario, it can handle both "cpu" and "memory" at the "task" level. Since we'll be releasing support for both of these with the same agent version, we don't need them to be 2 distinct capabilities.

Reply:
@aaithal thanks for the explanation.